sanic/router.py

```diff
@@ -163,7 +163,7 @@ def add_parameter(match):
             parameters.append(parameter)
 
             # Mark the whole route as unhashable if it has the hash key in it
-            if re.search(r'(^|[^^]){1}/', pattern):
+            if re.search(r'(^|[^^]){0}/', pattern):
```

Comment:
hmm what's going on here, shouldn't the pattern come first? https://docs.python.org/3/library/re.html#re.search